apps/web/src/pages/profile.tsx

```diff
@@ -34,59 +133,166 @@ const ProfilePage: NextPageWithLayout = () => {
+                            setIsRevokeApiKeyModalOpen(true);
+                            setApiKeyToRevoke(apiKeyId);
+                          }}
+                          className="bg-transparent text-2xl hover:bg-red-800/80"
+                        />
+                      </div>
+                    )
+                  )}
+                </>
+              )}
+
+              <form
+                className="pt-4"
+                onSubmit={(e) => {
+                  // TODO: Create API Key
+                  e.preventDefault();
+                }}
+              >
+                <label htmlFor="newApiKey" className=" font-semibold">
+                  Create a new API Key:
+                </label>
+                <div className="mt-2">
+                  <input
+                    ref={apiKeyNameRef}
+                    id="newApiKeyName"
+                    className="w-80 max-w-full rounded-l-md bg-gray-700 px-3 py-2 pr-2 focus:outline-none"
+                    placeholder="API Key name"
+                  />{" "}
+                  <button
+                    onClick={async () => {
+                      const name =
+                        apiKeyNameRef.current?.value ?? "New Api Key";
+
+                      const apiKey = await createApiKey({
+                        name: name,
+                        validDays: 365,
+                      });
+
+                      setNewApiKeyInfo({
+                        name: name,
+                        apiKey: apiKey,
+                      });
+                    }}
+                    className="-ml-1 mt-2 rounded-l-md rounded-r-md bg-gray-900 px-3 py-2 md:mt-0 md:rounded-l-none"
+                  >
+                    Create
+                  </button>
+                </div>
+              </form>
             </div>
-          </div>
-          <div>
-            <label htmlFor="name" className="block text-sm font-medium">
-              Email *
-            </label>
-            <div className="mt-1">
-              <Input type="text" defaultValue={data?.email ?? ""} disabled />
-            </div>
-          </div>
-          <div>
-            You signed up in via{" "}
-            <span className="font-medium capitalize underline underline-offset-2">
-              {data?.accounts[0]?.provider}
-            </span>
-          </div>
-          <DashboardButton type="submit" className="ml-auto block px-5 py-1">
-            Save
-          </DashboardButton>
-        </form>
+          </section>
+        </div>
       </div>
-    </div>
+      <CreateApiKeyModal
+        isOpen={isCreateApiKeyModalOpen}
+        apiKey={newApiKeyInfo?.apiKey ?? ""}
+        name={newApiKeyInfo?.name ?? ""}
+        onClose={() => {
+          setIsCreateApiKeyModalOpen(false);
+          setNewApiKeyInfo(null);
+        }}
+      />
+      <RevokeApiKeyModal
+        isOpen={isRevokeApiKeyModalOpen}
+        apiKey={apiKeyToRevoke ?? undefined}
```

Comment:
null to undefined? why do we use null at all?

Reply:
Good question about a philosophical question of javascript. I like having `null` to verbosely specify the explicit abstinence of a value. `undefined` is implicit version. I get why people like it the other way tho

Reply:
even JS uses undefined > null. array.find => undefined.

Reply:
so.. if you want to use null, do it everywhere! consistency is the key.